lint/builtin-shadow_checker.go

```diff
@@ -0,0 +1,66 @@
+package lint
+
+import (
+	"fmt"
+	"go/ast"
+)
+
+// BuiltinShadows detects when builtin functions shadowed in assignments
+type BuiltinShadows struct {
+	ctx *Context
+
+	warnings []Warning
+}
+
+// NewBuiltinShadows returns initialized checker for Go functions param names.
+func NewBuiltinShadows(ctx *Context) *BuiltinShadows {
+	return &BuiltinShadows{
+		ctx: ctx,
+	}
+}
+
+// Check runs builtin functions shadow check on assignments for f
+func (c *BuiltinShadows) Check(f *ast.File) []Warning {
+	c.warnings = c.warnings[:0]
+
+	builtinFuncs := map[string]struct{}{
+		"append":  struct{}{},
+		"cap":     struct{}{},
+		"close":   struct{}{},
+		"complex": struct{}{},
+		"copy":    struct{}{},
+		"delete":  struct{}{},
+		"imag":    struct{}{},
+		"len":     struct{}{},
+		"make":    struct{}{},
+		"new":     struct{}{},
+		"panic":   struct{}{},
+		"print":   struct{}{},
+		"println": struct{}{},
+		"real":    struct{}{},
+		"recover": struct{}{},
+	}
```

Comment:
If you don't like repeating `struct{}{}`, there are 2 options:
1. Use `map[string]bool`. That's less ugly.
2. Use this idiom:
```go
	builtins := []string{
		"append",
		"cap",
		"close",
		"complex",
		"copy",
		"delete",
		"imag",
		"len",
		"make",
		"new",
		"panic",
		"print",
		"println",
		"real",
		"recover",
	}
	builtinFuncs := make(map[string]struct{})
	for i := range builtins {
		builtinFuncs[builtins[i]] = struct{}{}
	}
```
But I'm OK with current approach.

Reply:
It's also better to initialize map literal inside constructor and store it inside checker.
This way, each `Check` call doesn't have to construct that map again.

Reply:
See `param-name` checker for example. It stores small map inside it.